### DISCUSSION TITLE
where should I deploy?

### DISCUSSION THREAD

**Discussion by marcomow** [nocodb/nocodb#5324] [Q&A] [answered]
Hi! 

I'm currently evaluating which would be the best option to deploy a nocodb instance. 
I've tested hosting on Railway and had no flaws with it. I see that deployment should also be quite straightforward with DigitalOcean, Elest, Heroku...but I can't get my head around which option to choose.

Which would be the cheapest?
With which is easiest to do backups?
What would be pros of other services?
Is there a standard recommendation?

**Comment by wingkwong** [comment, marked as answer]
It would be depending on your side. Currently we provide docker image and executables (in the future we'll have cloud version). DigitalOcean or Heroku would be more common. If you just wanna experience it first, you can also probably just host it locally. 

We suggest to have an external DB for storing the meta data, which configures in `NC_DB`. For backup, you can perform the backup there. If `NC_DB` is not defined, we'll create a sqlite file `noco.db` inside the container. In this case, you can mount and backup that single file.

**Comment by marcomow** [comment]
thank you, especially the backup part is way clearer now
